src/silo/query_engine/query_result.cpp

```diff
@@ -16,7 +16,11 @@ void to_json(nlohmann::json& json, const QueryResult& query_result) {
 // NOLINTNEXTLINE(readability-identifier-naming)
 void to_json(nlohmann::json& json, const QueryResultEntry& result_entry) {
    for (const auto& [field, value] : result_entry.fields) {
-      json[field] = value;
+      if (value.has_value()) {
+         json[field] = value.value();
+      } else {
+         json[field] = nlohmann::json();
+      }
```

Comment:
Would that work?
```suggestion
      json[field] = value.value_or(nlohmann::json());
```

Reply:
I experimented with this three lines quite a lot, but there was no easy way to set a field to null. Your proposed solution also makes the field an empty string (also in the float / int case) instead of null

Reply:
How is it different to the current solution then? Why does this yield "null"?

Reply:
The type of the statement is different I suppose. Could not quite figure it out myself. An explicit cast to another type or an embedded `to_json` somewhere could maybe do it, but I did not get it to work